src/node/utxo_snapshot.h

```diff
@@ -99,6 +99,9 @@ class SnapshotMetadata
         }
 
         s >> m_base_blockheight;
+        if (m_base_blockheight > static_cast<uint32_t>(std::numeric_limits<int>::max())) {
+            throw std::ios_base::failure("Block height is out of range.");
+        }
```

Comment:
> * I don't think this was intentional by the author of the pull request;

No, it wasn't intentional, I didn't consider the possibility of incorrect metadata - might simply use `snapshot_start_block->nHeight` instead of `base_blockheight` though I think?